pkg/provider/infisical/client.go

```diff
@@ -0,0 +1,128 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or impliec.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package infisical
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+
+	esv1beta1 "github.com/external-secrets/external-secrets/apis/externalsecrets/v1beta1"
+	"github.com/external-secrets/external-secrets/pkg/find"
+	"github.com/external-secrets/external-secrets/pkg/provider/infisical/api"
+)
+
+var errNotImplemented = errors.New("not implemented")
+
+// if GetSecret returns an error with type NoSecretError.
```

Comment:
Where is the return of `NoSecretError` in case of a secret not found?

Reply:
I do it from API level itself

https://github.com/external-secrets/external-secrets/pull/3477/files/5750fe178cebc37bd1056b817cb18250a1b75e08#diff-54d77c761cff754b43e7a53dc8911286a082d7b054b5647592abcff47aa285f7R202

Reply:
Totally missed this :)